runtimes/mlflow/mlserver_mlflow/metadata.py

```diff
@@ -0,0 +1,81 @@
+from typing import Union, Tuple, List
+
+from mlflow.types.schema import Schema, ColSpec, TensorSpec, DataType
+from mlflow.models.signature import ModelSignature
+
+from mlserver.settings import ModelSettings
+from mlserver.types import MetadataModelResponse, MetadataTensor, Tags
+from mlserver.codecs import NumpyCodec, StringCodec, Base64Codec, DatetimeCodec
+from mlserver.codecs.numpy import to_datatype
+
+InputSpec = Union[ColSpec, TensorSpec]
+
+DefaultInputPrefix = "input-"
+DefaultOutputPrefix = "output-"
+
+_MLflowToContentType = {
+    DataType.boolean: ("BOOL", NumpyCodec.ContentType),
+    DataType.integer: ("INT32", NumpyCodec.ContentType),
+    DataType.long: ("INT64", NumpyCodec.ContentType),
+    DataType.float: ("FP32", NumpyCodec.ContentType),
+    DataType.double: ("FP64", NumpyCodec.ContentType),
+    DataType.string: ("BYTES", StringCodec.ContentType),
+    DataType.binary: ("BYTES", Base64Codec.ContentType),
+    DataType.datetime: ("BYTES", DatetimeCodec.ContentType),
+}
+
+
+def _get_content_type(input_spec: InputSpec) -> Tuple[str, str]:
+    if isinstance(input_spec, TensorSpec):
+        datatype = to_datatype(input_spec.type)
+        content_type = NumpyCodec.ContentType
+        return datatype, content_type
+
+    # TODO: Check if new type, which may not exist
+    return _MLflowToContentType[input_spec.type]
+
+
+def _get_shape(input_spec: InputSpec) -> List[int]:
+    if isinstance(input_spec, TensorSpec):
+        return list(input_spec.shape)
+
+    return [-1]
```

Comment:
what would -1 mean? would downstream treat it as a special case? Should we set it as `None` instead?

Reply:
I see `-1` as the "batch dimension". I think Triton also uses the same convention. So in this case it would just mean a list of "shapeless things". We could use `[None]`, but I'm not sure whether / how that would get serialised on the metadata endpoints. 

For this case, do you think that maybe we should just return an empty list?